subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheFileSystemDefaultExcludesTest.groovy

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache
+
+import org.apache.tools.ant.DirectoryScanner
+import org.gradle.test.fixtures.file.TestFile
+
+class ConfigurationCacheFileSystemDefaultExcludesTest extends AbstractConfigurationCacheIntegrationTest {
```

Comment:
Could you please add some tests with composite builds - changing excludes in the `buildSrc`, or a plugin, or a "normal" included build.

Reply:
@mlopatkin Is there a scenario we have to add here?

Reply:
Besides the obvious one with nothing fancy, I'd start with the included build that modifies the set of excludes. AFAIR, its settings file should be evaluated later than the root one. BuildSrc I'd test separately too, as its lifecycle is slightly different. Another corner case would be modifying the set of excludes inside the `build.gradle` - if you don't hit the check in the PatternSpecFactory, you can get the new state into the configuration cache at least.

It would also be nice to add some coverage for Ant tasks, or modify existing Ant test cases to examine the configuration cache. The latter should happen automatically to some extent with load-after-store, but given that the set of excludes is the static state that is not reliably reset after store - it makes sense to test multiple build runs too.